content/rs/developing/crdbs/developing-streams-crdb.md

```diff
@@ -1,278 +1,320 @@
-1. Semi-strict: Similar to strict. XADD allows full-ID, but can create duplicate IDs.
+{{% note %}}
+Stream IDs in open source Redis consist of two integers separated by a dash ('-'). When the server generates the ID, the first integer is the current time in milliseconds, and the second integer is a sequence number. So, the format for stream IDs is MS-SEQ.
+{{% /note %}}
 
-The default and recommended mode is strict to prevent duplicate IDs. A stream with duplicate IDs can cause:
+To prevent duplicate IDs and to comply with the original Redis streams design, Active-Active databases provide three ID modes for XADD:
 
-1. XDEL, XCLAIM and other commands to affect more than one entry
-1. Trimmed duplicate entries in case of DB export and RENAME
+1. **Strict**: In _strict_ mode, XADD allows server-generated IDs (using the '`*`' ID specifier) or IDs consisting only of the millisecond (MS) portion. When the millisecond portion of the ID is provided, the ID's sequence number is calculated using the database's region ID. This prevents duplicate IDs in the stream. Strict mode rejects full IDs (that is, IDs containing both milliseconds and a sequence number).
+1. **Semi-strict**: _Semi-strict_ mode is just like _strict_ mode except that it allows full IDs (MS-SEQ). Because it allows full IDs, duplicate IDs are possible in this mode.
+1. **Liberal**: XADD allows any monotonically ascending ID. When given the millisecond portion of the ID, the sequence number will be set to `0`. This mode may also lead to duplicate IDs.
 
-### Iterating a stream with XREAD
+The default and recommended mode is _strict_, which prevents duplicate IDs
+
+{{% warning %}}
+Why do you want to prevent duplicate IDs? First, XDEL, XCLAIM, and other commands can affect more than one entry when duplicate IDs are present in a stream. Second, duplicate entries may be removed if a database is exported or renamed.
+{{% /warning %}}
+
+To change XADD's ID generation mode, use the `rladmin` command-line utility:
 
-Using XREAD to iterate through stream entries is a popular pattern and is similar to SCAN:
-
-```src
-127.0.0.1:6379> XADD x 110 f v
-"110-0"
-127.0.0.1:6379> XADD x 120 f v
-"120-0"
-127.0.0.1:6379> XADD x 130 f v
-"130-0"
-127.0.0.1:6379> XADD x 140 f v
-"140-0"
-127.0.0.1:6379> XADD x 150 f v
-"150-0"
-127.0.0.1:6379> XREAD COUNT 2 STREAMS x 0
-1) 1) "x"
-   2) 1) 1) "110-0"
-         2) 1) "f"
-            2) "v"
-      2) 1) "120-0"
-         2) 1) "f"
-            2) "v"
-127.0.0.1:6379> XREAD COUNT 2 STREAMS x 120
-1) 1) "x"
-   2) 1) 1) "130-0"
-         2) 1) "f"
-            2) "v"
-      2) 1) "140-0"
-         2) 1) "f"
-            2) "v"
-127.0.0.1:6379> XREAD COUNT 2 STREAMS x 140
-1) 1) "x"
-   2) 1) 1) "150-0"
-         2) 1) "f"
-            2) "v"
+Set _strict_ mode:
+```sh
+rladmin> tune db crdb crdt_xadd_id_uniqueness_mode strict
```

Comment:
I suggest writing it in the following format:
rladmin> tune db <crdb name> crdt_xadd_id_uniqueness_mode strict

Reply:
I find both confusing. I suggest:
`rladmin> tune myCRDB crdt_xadd_id_uniqueness_mode strict`